Test1/src/leetcode/LeetCode39.java

```diff
@@ -0,0 +1,48 @@
+package leetcode;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * https://leetcode.com/problems/combination-sum/description/
+ * @author jhm95
+ */
+public class LeetCode39 {
+	
+	List<List<Integer>> answerList = new ArrayList<>();
+
+	public List<List<Integer>> combinationSum(int[] candidates, int target) {
+		
+		findAnswer(candidates, new ArrayList<>(), target);
+
+		return answerList;
+	}
+
+	public void findAnswer(int[] candidates, List<Integer> answerSub, int target) {
+		if(target == 0) {
+			answerList.add(answerSub);
+			return;
+		}
+		List<Integer> answer;
+		int factor;
+		int rest;
+		for (int i = 0; i < candidates.length; i++) {
+			factor = candidates[i];
+			rest = target - factor;
+			if (rest >= 0 && !hasReverseOrder(answerSub, factor)) {
+                answer = new ArrayList<>(answerSub);
+				answer.add(factor);
+				findAnswer(candidates, answer, rest);
+			}
+		}
+	}
+
+	private boolean hasReverseOrder(List<Integer> answer, int factor) {

```

Comment:
Not quite sure what this is for. Can you explain this a bit?

Reply:
This functions is for removing duplicated case
For example, 
When the conditions are following case
Input: candidates = [2,3,6,7], target = 7
Output: [[2,2,3],[7]]

{2,2,3}, {2,3,2}, {3,2,2} those three cases are all allowed. 
To filter those cases, I make the hasReverseOrder method. 

Anyway, How about Germany?

Reply:
I guess you can try to avoid duplicated cases at the first place, then you don't need to remove anything in your code.

Germany is good so far. I love it very much! It's a pretty chilling place and I feel very relaxed every day.